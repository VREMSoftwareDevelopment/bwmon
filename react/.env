VITE_VERSION=$npm_package_version
VITE_NAME=$npm_package_name
VITE_DESCRIPTION=$npm_package_description
