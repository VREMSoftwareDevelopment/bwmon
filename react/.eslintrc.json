{
    "extends": [
        "eslint:recommended",
        "plugin:import/errors",
        "plugin:react/recommended",
        "plugin:jsx-a11y/recommended",
        "plugin:prettier/recommended"
    ],
    "plugins": ["react", "import", "jsx-a11y", "jest", "prettier"],
    "settings": {
        "react": {
            "version": "detect"
        }
    },
    "rules": {
        "react/prop-types": "off",
        "indent": ["error", 4],
        "linebreak-style": "error",
        "quotes": ["error", "single"],
        "no-unused-vars": [
            "error",
            {
                "vars": "all",
                "varsIgnorePattern": "^_",
                "args": "after-used",
                "argsIgnorePattern": "^_",
                "caughtErrors": "all",
                "caughtErrorsIgnorePattern": "^_"
            }
        ]
    },
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module",
        "ecmaFeatures": {
            "jsx": true
        }
    },
    "env": {
        "es6": true,
        "browser": true,
        "node": true,
        "jest/globals": true
    }
}
