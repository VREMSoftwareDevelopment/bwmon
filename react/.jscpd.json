{
    "minLines": 10,
    "minTokens": 50,
    "reporters": ["console", "html"],
    "output": "reports/duplication",
    "ignore": ["**/*.test.js", "**/*.test.jsx"]
}
