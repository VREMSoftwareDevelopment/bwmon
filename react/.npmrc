save=true
puppeteer_skip_download=true
PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true