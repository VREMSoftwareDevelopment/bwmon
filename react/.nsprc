{
    "1070415": {
        "active": true,
        "expiry": "2022-12-31",
        "notes": "Ignored - waiting for update"
    }
}
