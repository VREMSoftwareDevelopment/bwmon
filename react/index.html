<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <link rel="icon" href="/favicon.ico" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
        <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />
        <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width" />
        <meta name="theme-color" content="#000000" />
        <meta name="Bandwidth Monitor" content="Bandwidth MonitorApplication" />
        <link rel="manifest" href="/manifest.json" />
        <base href="/" />
        <title>Bandwidth Monitor</title>
    </head>
    <body>
        <noscript>You need to enable JavaScript to run this app.</noscript>
        <div id="root"></div>
        <script type="module" src="/src/index.jsx"></script>
    </body>
</html>
