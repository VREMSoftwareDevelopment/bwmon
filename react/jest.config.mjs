// jest.config.mjs
export default {
    testEnvironment: '<rootDir>/jest.enviroment.js',
    testMatch: ['<rootDir>/src/**/*.test.js?(x)'],
    modulePaths: ['<rootDir>/src/'],
    transform: {
        '^.+\\.[t|j]sx?$': 'babel-jest',
    },
    collectCoverageFrom: ['src/**/*.{js,jsx}', '!src/index.jsx', '!src/serviceWorker.js', '!**/e2e/**', '!**/__mocks__/**'],
    coverageThreshold: {
        global: {
            branches: 99,
            functions: 93,
            lines: 99,
            statements: 97,
        },
    },
};
