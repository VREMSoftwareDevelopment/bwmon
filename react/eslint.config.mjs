import { fixupConfigRules, fixupPluginRules } from '@eslint/compat';
import react from 'eslint-plugin-react';
import _import from 'eslint-plugin-import';
import jsxA11Y from 'eslint-plugin-jsx-a11y';
import jest from 'eslint-plugin-jest';
import prettier from 'eslint-plugin-prettier';
import globals from 'globals';
import path from 'node:path';
import { fileURLToPath } from 'node:url';
import js from '@eslint/js';
import { FlatCompat } from '@eslint/eslintrc';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const compat = new FlatCompat({
    baseDirectory: __dirname,
    recommendedConfig: js.configs.recommended,
    allConfig: js.configs.all,
});

export default [
    ...fixupConfigRules(
        compat.extends(
            'eslint:recommended',
            'plugin:import/errors',
            'plugin:react/recommended',
            'plugin:jsx-a11y/recommended',
            'plugin:prettier/recommended'
        )
    ),
    {
        plugins: {
            react: fixupPluginRules(react),
            import: fixupPluginRules(_import),
            'jsx-a11y': fixupPluginRules(jsxA11Y),
            jest,
            prettier: fixupPluginRules(prettier),
        },
        languageOptions: {
            globals: {
                ...globals.browser,
                ...globals.node,
                ...jest.environments.globals.globals,
            },
            ecmaVersion: 'latest',
            sourceType: 'module',
            parserOptions: {
                ecmaFeatures: {
                    jsx: true,
                },
            },
        },
        settings: {
            react: {
                version: 'detect',
            },
        },
        rules: {
            'react/prop-types': 'off',
            indent: ['error', 4],
            'linebreak-style': 'off',
            quotes: ['error', 'single'],
            'no-unused-vars': [
                'error',
                {
                    vars: 'all',
                    varsIgnorePattern: '^_',
                    args: 'after-used',
                    argsIgnorePattern: '^_',
                    caughtErrors: 'all',
                    caughtErrorsIgnorePattern: '^_',
                },
            ],
        },
    },
];
