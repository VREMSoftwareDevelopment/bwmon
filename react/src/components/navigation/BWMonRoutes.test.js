/*
 *      Copyright (C) 2010 - 2024 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import BWMonRoutes from './BWMonRoutes';

const MockComponent = ({ text }) => <div>{text}</div>;

describe('BWMonRoutes', () => {
    const menu = [
        { pathname: '/page1', element: <MockComponent text="Page 1" /> },
        { pathname: '/page2', element: <MockComponent text="Page 2" /> },
    ];

    it('renders default route', () => {
        render(
            <MemoryRouter initialEntries={['/']}>
                <BWMonRoutes menu={menu} />
            </MemoryRouter>
        );
        expect(screen.getByText('Page 1')).toBeInTheDocument();
    });

    it('renders specific route', () => {
        render(
            <MemoryRouter initialEntries={['/page2']}>
                <BWMonRoutes menu={menu} />
            </MemoryRouter>
        );
        expect(screen.getByText('Page 2')).toBeInTheDocument();
    });

    it('renders PageNotFound for unknown route', () => {
        const { container } = render(
            <MemoryRouter initialEntries={['/page3']}>
                <BWMonRoutes menu={menu} />
            </MemoryRouter>
        );
        expect(container.firstChild).toBeNull();
    });
});
