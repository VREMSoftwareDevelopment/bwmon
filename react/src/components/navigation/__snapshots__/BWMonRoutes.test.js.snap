// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BWMonRoutes renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1dtoofl-MuiPaper-root"
  style={
    Object {
      "--Paper-shadow": "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
    }
  }
>
  <div
    className="makeStyles-root-1"
  >
    <span
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-lrbo11-MuiLinearProgress-root"
      role="progressbar"
    >
      <span
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-eg0r9f-MuiLinearProgress-bar1"
        style={Object {}}
      />
      <span
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1yf7653-MuiLinearProgress-bar2"
        style={Object {}}
      />
    </span>
    <div
      className="makeStyles-root-2"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorInfo MuiAlert-standardInfo MuiAlert-standard css-rxaxkm-MuiPaper-root-MuiAlert-root"
        role="alert"
        style={
          Object {
            "--Paper-shadow": "none",
          }
        }
      >
        <div
          className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1lzspe0-MuiSvgIcon-root"
            data-testid="InfoOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
        >
          <div
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-1hs5ppg-MuiTypography-root-MuiAlertTitle-root"
            style={Object {}}
          >
            Loading...
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  />
</div>
`;
